File: app/src/main/java/com/silentgames/silent_planet/model/entities/space/SpaceShip.java
package com.silentgames.silent_planet.model.entities.space;

import com.silentgames.silent_planet.model.entities.EntityType;
import com.silentgames.silent_planet.model.entities.EntityTypeEx;
import com.silentgames.silent_planet.model.entities.ground.PlayersOnCell;

/**
 * Created by gidroshvandel on 07.07.16.
 */
public class SpaceShip extends EntityTypeEx {

    private int crystals = 0;

    public int getCrystals() {
        return crystals;
    }

    public void setCrystals(int crystals) {
        this.crystals = crystals;
    }
}

File: app/src/main/java/com/silentgames/silent_planet/model/entities/space/SpaceShip.kt
package com.silentgames.silent_planet.model.entities.space

import android.content.Context
import com.silentgames.silent_planet.model.entities.EntityType
import com.silentgames.silent_planet.model.entities.ground.Player
import com.silentgames.silent_planet.model.fractions.Fractions

/**
 * Created by gidroshvandel on 07.07.16.
 */
abstract class SpaceShip(
        context: Context,
        fraction: Fractions,
        var playersOnBord: MutableList<Player> = mutableListOf(),
        override var isCanFly: Boolean = true,
        override var isCanMove: Boolean = false
) : EntityType(context, fraction)
